src/TestUtils/CloudFunctionDeploymentTrait.php

```diff
@@ -101,6 +89,12 @@ private static function doDelete()
      */
     public function setUpClient()
     {
+        // CloudEvent functions do not have URLs.
+        // Guzzle client creation stalls without a valid URL.
+        if (self::$isCloudEventFunction === true) {
```

Comment:
I don't like that this variable could possibly not exist. Can we add this as `private static` in the test trait and then the TestCase classes can set it to true in their `setUp` methods?

Also, I think this would be better as `$functionSignatureType`, and we just check if it's `cloudevent`...

Reply:
This is also the pattern for $this->name, will adjust traits for both.